### ISSUE TITLE
typeorm-typescript/enforce-relation-types, support "Relation<Other>"

### ISSUE THREAD

**Issue by norbornen** [daniel7grant/eslint-plugin-typeorm-typescript#17]
Hello!

Can u add support for this case:
```
import { Column, Entity, OneToOne, OneToMany, Relation } from 'typeorm';

class Entity {
    // Should be Other | null
    @OneToOne(() => Other)
    @JoinColumn()
    other: Relation<Other>;

    @OneToMany(() => Other, ...)
    other2: Relation<Other>[];
}
```

https://github.com/typeorm/typeorm/issues/4526#issuecomment-1076462998

This is the correct description of the column, but the linter reports an error:
```
Type of other in Entity is not consistent with the TypeORM relation type OneToMany
(expected type: Other).eslint[typeorm-typescript/enforce-relation-types](https://github.com/daniel7grant/eslint-plugin-typeorm-typescript#typeorm-typescriptenforce-relation-types)
```

Thank you for your work!

**Comment by daniel7grant** [comment]
Thank you for the issue, there is already an ongoing PR in the form of #16 about adding a relation wrapper. I didn't have time so far to look over the implementation but I will try to do it in the following days.

**Comment by daniel7grant** [comment]
It turned out to be a smaller change than expected, I added this in #18. Feel free to test it and I'll put out a new patch version in the following days.

**Comment by norbornen** [comment]
And, just so you know, the plugin does not work correctly if `"strictNullChecks": false` in project. I am sure that this is the problem of those who use it, but I think it would be right to write this to README.

**Comment by daniel7grant** [comment]
Thanks for testing!

As for the order of the wrapper, I've chosen somewhat arbitrarily `Relation<T[]>` , mostly based on this [Stack Overflow thread](https://stackoverflow.com/questions/74018663/typeorm-how-do-you-use-the-ts-relation-type-for-one-to-many-relations). I couldn't find anything in the TypeORM docs, so I just used this answer. If you can point me towards any official suggestion, I'll change it.

I tried with `OneToMany`, it seems to be working to me. Your problem might be that if you misspell the name, it won't be added to the suggestion either: `d: Relation<Aaa>` will suggest `d: Relation<Aaa[]>` correctly.

For `strictNullChecks`, I wasn't able to reproduce it. Can you send me a tsconfig or anything that makes it break with the [example configuration](https://github.com/daniel7grant/eslint-plugin-typeorm-typescript/tree/master/examples/recommended)? Assuming this is a real problem, I'm not sure how to deal with it. I'll probably follow your advice and add a note to the README... without strict null-checking, this is mostly a cosmetic library.

**Comment by daniel7grant** [comment]
I merged the wrapper configuration in #18 with this setup. If you have any more issues feel free to reopen.
